io/msgpack: Raise a more informative exception on file not found

smspillaz opened the pull request:
- [x] closes #27160
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

jreback commented on pandas/tests/io/test_packers.py:
```diff
@@ -934,3 +934,7 @@ def test_msgpack_period_freq(self):
         repr(r)
+
+    def test_msgpack_file_not_found(self):
+        bad_path = '1/2/3/4/path/does/not/exist'
```
can you add the issue number here as a comment

smspillaz replied:
I removed this test and added the issue number inline where the fix was done.

jreback reviewed (changes requested):
can you add a whatsnew note in the bug fixes IO section, otherwise lgtm. CI is currently borked on 1 build, but ping when rest are green.

simonjayhawkins commented on pandas/tests/io/test_packers.py:
```diff
@@ -934,3 +934,7 @@ def test_msgpack_period_freq(self):
+
+    def test_msgpack_file_not_found(self):
+        bad_path = '1/2/3/4/path/does/not/exist'
+        pytest.raises(read_msgpack, FileNotFoundError, bad_path)
```
use context manager or better still, i don't think you need to add a test.

 changing `TestCommonIOCapabilities.test_read_non_existant` and `TestCommonIOCapabilities.test_read_expands_user_home_dir` in `pandas/tests/io/test_common.py` would be sufficient.

smspillaz replied:
Ah, nice - looks like these tests handle this behavior already.

simonjayhawkins commented on pandas/io/packers.py:
```diff
@@ -184,6 +184,8 @@ def read(fh):
             with open(path_or_buf, 'rb') as fh:
                 return read(fh)
+        else:
+            return FileNotFoundError('{} not found'.format(path_or_buf))
```
although this may simplify testing, i'd be inclined to just let Python raise the system specific exception. 

i.e.

replace
```python
    # see if we have an actual file
    if isinstance(path_or_buf, str):
        try:
            exists = os.path.exists(path_or_buf)
        except (TypeError, ValueError):
            exists = False

        if exists:
            with open(path_or_buf, 'rb') as fh:
                return read(fh)
```

with
```python
    if isinstance(path_or_buf, str):
        with open(path_or_buf, 'rb') as fh:
            return read(fh)
```

and it raises (on windows)..
``` 
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\simon\\OneDrive\\code\\pandas-simonjayhawkins\\pandas\\tests\\io\\data\\does_not_exist.mp'
```

smspillaz replied:
Agreed

Commit:
io/packers: Just let python raise itself if the file was not found

As opposed to checking if it exists first.

smspillaz commented:
Okay, requested fixes should be in place now:
 - Fixed CI
 - Removed test and instead amended `TestIOCommon`
 - Added `whatsnew` entry
 - Added comment in `packers.py` indicating bug

WillAyd commented:
@smspillaz looks like there are some merge conflicts - can you clean those up?

smspillaz commented:
Can be closed as #27201 was merged which fixes the same issue
